src/metadataGeneration/typeResolver.ts

```diff
@@ -254,6 +254,29 @@ export class TypeResolver {
         return stringMetaType;
       }
 
+      const type = this.current.typeChecker.getTypeFromTypeNode(this.typeNode);
+      const toJSON = this.current.typeChecker.getPropertyOfType(type, 'toJSON');
```

Comment:
We should not use the typeChecker by default. Best way is always to recursively call the TR on the typeNode and only fall back to the inferred type if there's no explicit typeNode.